### ISSUE TITLE
Switch workspace by scroll action is broken

### ISSUE THREAD

**Issue by AlexWayfer** [micheleg/dash-to-dock#710]
Works when "Show applications" button is hovered, doesn't work at any another icon.

Latest extensions.gnome.org release, 62.

**Comment by AlexWayfer** [comment]
Please, it's a critical bug.

@micheleg, maybe you want to find a new maintainer? You can add a note about this into README.

**Comment by micheleg** [comment]
I think this is the intended behaviour. Do you have the icons size "fixed" in the settings? In such case to avoid collision between the different scroll actions, we enable the workspace switching only on the show applications button.

**Comment by AlexWayfer** [comment]
> I think this is the intended behaviour.

What? The description is:

> Behavior when scrolling on the icon of an application.

Not "Show applications" button.

> Do you have the icons size "fixed" in the settings?

Yes.

> In such case to avoid collision between the different scroll actions, we enable the workspace switching only on the show applications button.

Oh, I see now.

1. What about cases when all icons are showed? There is no scroll for revealing them.
2. I think it should behaves like a scroll of element, `<textarea>` for example, in a HTML page: firstly scroll element, then document (it's parent).

**Comment by micheleg** [comment]
Yes, the description can indeed be improved. We ideally need to find a way to make it clear the functionalities are mutually exclusive. I think your suggestion make logically sense but would be a bit confusing for the user.

**Comment by AlexWayfer** [comment]
> I think your suggestion make logically sense but would be a bit confusing for the user.

What? No! It's standard situation and behavior for many places, around the whole web and mobile applications.

Look: I have `textarea` right here, and it has scrollbar. I'm scrolling — textarea is scrolled — and then, at the end of textarea's scrollbar, scroll propagate to the document.

**Comment by micheleg** [comment]
But in that case you are achieving the same goal: you are asking for "show me more", and the browser shows you more. In our case you are mixing together two different actions: "show me more icons" and "switch workspace", which are totally uncorrelated. And once you have fired off the wrong action (unexpectedly switched to the next workspace) you need to scroll back to the top of the dock to then go back. This is my feeling, but yes, I would need to try to implement this and test it to really get a final opinion, and unfortunately at the moment I have not much time to experiment on it. Help is very much welcome, both on the feature implementation and settings explanation.

**Comment by AlexWayfer** [comment]
> But in that case you are achieving the same goal: you are asking for "show me more", and the browser shows you more. In our case you are mixing together two different actions: "show me more icons" and "switch workspace", which are totally uncorrelated.

Oh, OK, I got it. But for me "Switch workspace" by scrolling is like "Show me more, show me another workspace". :grimacing:

**Comment by flega68** [comment]
I think the perfect solution was in version 61. 
With fixed icon size:
When I hover over a icon, the mouse wheel scrolls the icons,
When I touch the screen border with the mouse, it  switches the workspaces.

**Comment by AlexWayfer** [comment]
> I think the perfect solution was in version 61.With fixed icon size:When I hover over a icon, the mouse wheel scrolls the icons,When I touch the screen border with the mouse, it switches the workspaces.

That sounds not bad.

**Comment by flega68** [comment]
Optional would be nice. What is the but...?

**Comment by eassymo** [comment]
please enable this option :D

**Comment by flega68** [comment]
Until today I helped myself with staying on Version 61. I upgraded to Fedora 29 and now Version 61 doesn't work anymore. And still no Button to activate the old behaviour :(
Pretty Please :)

**Comment by flega68** [comment]
I found a solution to my problem. I installed  the extension "Scroll Workspace". Now I have my old functionality back. (I had to put   Scroll Workspace on the right site and fix a minor bug in it, but that is a different story)

**Comment by dvorapa** [comment]
I can still reproduce this

**Comment by mitar** [comment]
+1 to allow scrolling on 1px edge again.

**Comment by mitar** [comment]
Also, for some reason scrolling on "show applications" does not switch workspaces for me on Ubuntu Docker on Ubuntu 20.04. `scroll-switch-workspace` is set to true.

**Comment by flega68** [comment]
Switching on "show applications" works for me (Fedora 32, DashtoDock 68). But I still miss the good old times where I was able to switch the workspaces on the whole left side ;)
So: +1 to allow scrolling on 1px edge
